go/vt/vtgate/planbuilder/testdata/memory_sort_cases.json

```diff
@@ -655,5 +655,55 @@
         "user.user"
       ]
     }
+  },
+  {
+    "comment": "Derived table split across two shards, and ordered by both",
+    "query": "select * from (select u.foo, ue.bar from user u, user_extra ue) tbl order by tbl.bar, tbl.foo",
+    "plan": {
+      "QueryType": "SELECT",
+      "Original": "select * from (select u.foo, ue.bar from user u, user_extra ue) tbl order by tbl.bar, tbl.foo",
+      "Instructions": {
+        "OperatorType": "Sort",
+        "Variant": "Memory",
+        "OrderBy": "(1|2) ASC, (0|3) ASC",
+        "ResultColumns": 2,
+        "Inputs": [
+          {
+            "OperatorType": "Join",
```

Comment:
nitpicking here, but a better plan would push the `tbl.bar` column to the LHS, and then we only need to do memory sort on the inputs from the RHS. We would not need to sort the full result set with that plan